magic_the_gathering/game_state.py

```diff
@@ -68,18 +76,23 @@ def set_decks(self, decks: List[List[Card]]):
         self.decks = decks
 
     def cast_card(self, player_index, hand_card_index):
+        # TODO: Should we implement a play_card method? Is there a difference between casting and playing a card?
         card = self.hands[player_index].pop(hand_card_index)
-        self.stack.append(StackItem(card=card, owner_player_index=player_index))
+        self.stack.append(card)
 
     def resolve_top_of_the_stack(self):
-        stack_item = self.stack.pop(-1)
-        board_item = BoardItem(card=stack_item.card, state=CardPosition.TAPPED)
-        self.boards[stack_item.owner_player_index].append(board_item)
+        # TODO: Currently, this was implemented with casting creatures or lands in mind. It should be generalized.
+        card = self.stack.pop(-1)
+        card.is_tapped = True
```

Comment:
Even if we're just talking lands and creatures, they do not enter tapped unless specified.
Also, a land isn't cast and therefore does not go on the stack (this is one of the rare game actions that do not use the stack). Also it means that players cannot respond to a land being played, it just happens instantaneously.

Reply:
The definition of summoning sickness is that the creature cannot attack or be tapped as part of the cost of one of its activated abilities unless you controlled it continuously since the beginning of your turn (did not check actual definition, might be some subtle difference).
So to track this we should probably have an attribute per permanent (because some non-creatures can turn into creatures) like `started_turn_controlled_by` that can be null to mean no one, or be a player index. Starts at null. At the beginning of each turn, we set this flag for each permanent. If a permanent leaves the battlefield it and ends up back on the battlefield, it's a new object, it has no memory of its previous state

Reply:
> Even if we're just talking lands and creatures, they do not enter tapped unless specified

Ok I fixed it, now I set `is_tapped` to `False` (see last commit)

Reply:
Now, regarding your other comments, I think we will discuss it together later to find the best solutions :slightly_smiling_face: